src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceEventGenerator.cs

```diff
@@ -0,0 +1,68 @@
+
+        public static string MetricEventRegex { get; } = $"(?<SubscriptionId>[^,]*),(?<AppName>[^,]*),(?<FunctionName>[^,]*),(?<EventName>[^,]*),(?<Average>\\d*),(?<Min>\\d*),(?<Max>\\d*),(?<Count>\\d*),(?<HostVersion>[^,]*),(?<EventTimestamp>[^,\"]+)";
+
+        public static string DetailsEventRegex { get; } = $"(?<AppName>[^,]*),(?<FunctionName>[^,]*),\"(?<InputBindings>.*)\",\"(?<OutputBindings>.*)\",(?<ScriptType>[^,]*),(?<IsDisabled>[0|1])";
+
+        public void LogFunctionTraceEvent(LogLevel level, string subscriptionId, string appName, string functionName, string eventName,
+            string source, string details, string summary, string exceptionType, string exceptionMessage,
+            string functionInvocationId, string hostInstanceId, string activityId)
+        {
+            var eventTimestamp = DateTime.UtcNow.ToString(EventTimestampFormat);
+            var hostVersion = ScriptHost.Version;
+            FunctionsSystemLogsEventSource.Instance.SetActivityId(activityId);
+
+            var logger = _loggerFactory.GetOrCreate(FunctionsLogsCategory);
+            WriteEvent(logger, $"{(int)ToEventLevel(level)},{subscriptionId},{appName},{functionName},{eventName},{source},{NormalizeString(details)},{NormalizeString(summary)},{hostVersion},{eventTimestamp},{exceptionType},{NormalizeString(exceptionMessage)},{functionInvocationId},{hostInstanceId},{activityId}");
+        }
+
+        public void LogFunctionMetricEvent(string subscriptionId, string appName, string functionName, string eventName, long average,
+            long minimum, long maximum, long count, DateTime eventTimestamp)
+        {
+            var hostVersion = ScriptHost.Version;
+            var logger = _loggerFactory.GetOrCreate(FunctionsMetricsCategory);
+            WriteEvent(logger, $"{subscriptionId},{appName},{functionName},{eventName},{average},{minimum},{maximum},{count},{hostVersion},{eventTimestamp.ToString(EventTimestampFormat)}");
+        }
+
+        public void LogFunctionDetailsEvent(string siteName, string functionName, string inputBindings, string outputBindings,
+            string scriptType, bool isDisabled)
+        {
+            var logger = _loggerFactory.GetOrCreate(FunctionsDetailsCategory);
+            WriteEvent(logger, $"{siteName},{functionName},{NormalizeString(inputBindings)},{NormalizeString(outputBindings)},{scriptType},{(isDisabled ? 1 : 0)}");
+        }
+
+        public void LogFunctionExecutionAggregateEvent(string siteName, string functionName, long executionTimeInMs,
+            long functionStartedCount, long functionCompletedCount, long functionFailedCount)
+        {
+            throw new NotImplementedException();
```

Comment:
The execution pipeline will be calling into this and the one below - won't these exceptions cause things to blow up?

Reply:
Fixed.